### PR TITLE
Basic functionality for Corelli calibration database

### PR DESCRIPTION
**Pull request by williamfgc**
**Description of work.**

Added a function to allow combining calibrated `TableWorkspaces` in a `WorkspaceGroup` into a single `TableWorkspace`
Valid for ORNL Corelli instrument
Added unit test

  @jmborr 

**To test:**

Unit tests added to CI `test_corelli_calibration.py`

Part of efforts to support Corelli instrument calibration, this is work in progress

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by jmborr — CHANGES_REQUESTED** [review]

File: scripts/test/corelli/calibration/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 set(TEST_PY_FILES
     test_utils.py
+    test_corelli_calibration.py
```
[the remark this answers is not in the document]
Comment (williamfgc) [reply scripts/test/corelli/calibration/CMakeLists.txt:5]: sounds good, I was having issues with the `setUp / tearDown` functions

[the remark this answers is not in the document]
Comment (williamfgc) [reply scripts/test/corelli/calibration/CMakeLists.txt:5]: sounds good, what about `calibration_data.py`?

File: scripts/corelli/calibration/utils.py [hunk 2]
```diff
@@ -57,3 +59,56 @@ def calculate_tube_calibration(workspace: InputWorkspace, tube_name: str, shadow
     fit_par = TubeCalibFitParams(wire_pixel_positions, height=peak_height, width=peak_width, margin=fit_domain)
     fit_par.setAutomatic(True)
     return tube.calibrate(workspace, tube_name, wire_positions, peaks_form, fitPar=fit_par)
+
+
+def combine_spatial_banks(tables:WorkspaceGroup)->TableWorkspace:
```
[the remark this answers is not in the document]
Comment (williamfgc) [reply scripts/corelli/calibration/utils.py]: Let's push this for a future PR. I need to learn how the memory is managed when going with that option. Ideally, I'd like to limit the user choices.

File: scripts/test/corelli/calibration/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (williamfgc) [reply scripts/test/corelli/calibration/CMakeLists.txt]: Nevermind, we settled offline with `database.py`

**Comment by williamfgc** [comment]
@jmborr I restructured the files and functionality as discussed from your previous review. It's ready for your review. Thanks!

**Comment by williamfgc** [comment]
@peterfpeterson Windows build is retriggered

File: scripts/test/corelli/calibration/test_database.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+from corelli.calibration.database import combine_spatial_banks, init_corelli_table, save_bank_table, filename_bank_table,\
+    load_bank_table, has_valid_columns
+
+
+class TestCorelliDatabase(unittest.TestCase):
+
+    def setUp(self) -> None:
+
+        # create a mock database and tests save_bank_table and load_bank_table
+        self.database_path:str = 'data/corelli'
```
Comment (peterfpeterson) [inline scripts/test/corelli/calibration/test_database.py:27]: You should use [tempfile.NamedTemporaryFile](https://docs.python.org/3.6/library/tempfile.html#tempfile.NamedTemporaryFile) for where to save the file. I think you are writing in an illegal place Something like
```python
self.database_path:str = NamedTemporaryFile(prefix='corelli', delete=False).name
```
will likely work. Don't forget to delete the file in your `cleanup` step.
filename = NamedTemporaryFile

Comment (williamfgc) [reply scripts/test/corelli/calibration/test_database.py]: OK, thanks. I end up using [pathlib.TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) as a static member. It seems to be called every time I reuse it to create a file (interesting behavior). All files and directories should be cleaned at `tearDown`. Let see if it passes this time. Thanks!

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**